ast/dml.go

```diff
@@ -1060,7 +1060,8 @@ type SelectStmt struct {
 	// TableHints represents the table level Optimizer Hint for join type
 	TableHints []*TableOptimizerHint
 	// IsInBraces indicates whether it's a stmt in brace.
-	IsInBraces bool
+	IsInBraces       bool
+	WithBeforeBraces bool
```

Comment:
it will be useful if add a comment for `WithBeforeBraces`

Reply:
Do you mean parentheses `()`? I remember brace means `{}`.

Reply:
IsInBraces here refer to the `()` too (why do we need to care if `()` exists anyway). Legacy typo 🤷

Reply:
It's used to distinguish (with xxx select xxx) and with xxx (select xxx)